clientconn_test.go

```diff
@@ -85,6 +85,18 @@ func TestTLSServerNameOverwrite(t *testing.T) {
 	}
 }
 
+func TestWithAuthority(t *testing.T) {
```

Comment:
Can you also test that `WithAuthority` has no effect if `TransportCredentials` are present?
I think just copying `TestTLSServerNameOverwrite` and add `WithAuthority` to `Dial` would do the work.